app/com/linkedin/drelephant/spark/heuristics/UnifiedMemoryHeuristic.scala

```diff
@@ -116,10 +116,14 @@ object UnifiedMemoryHeuristic {
       }
     }.max
 
-    lazy val severity: Severity = if (sparkExecutorMemory <= MemoryFormatUtils.stringToBytes(unifiedMemoryHeuristic.sparkExecutorMemoryThreshold)) {
-      Severity.NONE
+     lazy val severity: Severity = if (sparkMemoryFraction > 0.05D && maxMemory > 268435456L) {
+      if (sparkExecutorMemory <= MemoryFormatUtils.stringToBytes(unifiedMemoryHeuristic.sparkExecutorMemoryThreshold)) {
+        Severity.NONE
+      } else {
+        PEAK_UNIFIED_MEMORY_THRESHOLDS.severityOf(maxUnifiedMemory)
```

Comment:
It would be great if you can add the points mentioned the pull request's description as comments in the code.